code/modules/lootpanel/loot_procs.dm

```diff
@@ -0,0 +1,171 @@
+/// Converts the contents of the tile into search objects
+/datum/lootpanel/proc/convert_tile_contents(list/slice)
+	var/list/found = list()
+
+	for(var/atom/thing as anything in slice)
+		if(thing.mouse_opacity == MOUSE_OPACITY_TRANSPARENT)
+			continue
+		if(thing.IsObscured())
+			continue
+		if(thing.invisibility > user.see_invisible)
+			continue
+
+		var/ref = REF(thing)
+		if(contents[ref])
+			continue
+
+		RegisterSignals(thing, list(
+			COMSIG_MOVABLE_MOVED,
+			COMSIG_QDELETING,
+			), PROC_REF(on_item_moved))
+
+		var/datum/search_object/item = new(user, thing)
+		found[ref] = item
```

Comment:
if we register onto QDELETING why do we need a ref for this? also won't it collide?

Reply:
is it for the ui?

Reply:
yes, basically I maintain a list of search_objects in `contents` to display to the ui. if these items are moved or deleted the ui updates